src/pde_solvers/polarpoissonlikesolver.hpp

```diff
@@ -636,15 +642,20 @@ class PolarSplineFEMPoissonLikeSolver
                 const Idx<BSplinesTheta> idx_trial_theta(ddc::select<BSplinesTheta>(idx_trial));
 
                 // Find the index range covering the cells where both the test and trial functions are non-zero
-                const Idx<RCellDim> first_overlap_element_r(
-                        idx_trial_r.uid() < BSplinesR::degree()
-                                ? 0
-                                : idx_trial_r.uid() - BSplinesR::degree());
+                const std::size_t idx_trial_r_val = (idx_trial_r - m_idx_bsplines_r_front).value();
+                const std::size_t idx_trial_theta_val
+                        = (idx_trial_theta - m_idx_bsplines_theta_front).value();
+                const std::size_t first_overlap_element_r_val
+                        = idx_trial_r_val < BSplinesR::degree()
+                                  ? 0
+                                  : idx_trial_r_val - BSplinesR::degree();
+
+                const Idx<RCellDim> first_overlap_element_r(first_overlap_element_r_val);
```

Comment:
Are integers needed here? Is it not possible to do something like:
```suggestion
const Idx<RCellDim> first_overlap_element_r(
                        idx_trial_r < Idx<RCellDim>(BSplinesR::degree())
                                ? Idx<RCellDim>(0)
                                : idx_trial_r - BSplinesR::degree());
```
?
Or is `idx_trial_r` the wrong type for this?

Reply:
Indeed, `idx_trial_r` is a `Idx<BSplinesR>` and `first_overlap_element_r` a `Idx<RCellDim>`.

Reply:
Done in https://github.com/gyselax/gyselalibxx/pull/196/commits/9092bfe093b512a5f0005863e77264a7a001ad16